x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java

```diff
@@ -368,8 +373,13 @@ private static EsField createField(String fieldName, Map<String, Map<String, Fie
 
         return esField;
     }
-    
+
     private static EsField createField(String fieldName, String typeName, Map<String, EsField> props, boolean isAggregateable) {
+        return createField(fieldName, typeName, props, isAggregateable, false);
+    }
+    
+    private static EsField createField(String fieldName, String typeName, Map<String, EsField> props, boolean isAggregateable,
+            boolean isSynthetic) {
```

Comment:
It seems you are not doing anything with `isSynthetic`? Leftover from a previous attempt?